lib/compress/index.js

```diff
@@ -6974,14 +6984,16 @@ def_optimize(AST_Conditional, function(self, compressor) {
     }
 
     // a == null ? b : a -> a ?? b
+    // a != null ? a : b -> a ?? b
     if (
         compressor.option("ecma") >= 2020 &&
-        is_nullish_check(condition, alternative, compressor)
+        is_nullish_check(condition, consequent, alternative, compressor)
     ) {
+        let left = condition.operator == "==" || condition.operator == "||" ? alternative : consequent;
         return make_node(AST_Binary, self, {
             operator: "??",
-            left: alternative,
-            right: consequent
+            left: left,
+            right: left == alternative ? consequent : alternative,
         }).optimize(compressor);
```

Comment:
I guess is_nullish_check could return the left and right hand side:

```
const nullish_check = get_nullish_check(condition, consequent, alternative, compressor);
if (nullish_check != null) { // very meta line here
  const [left, right] = nullish_check
  // create "??" expression using left and right
}
```

Reply:
Done.